nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/sensitive/aws/kms/AWSKMSSensitivePropertyProviderIT.groovy

```diff
@@ -0,0 +1,253 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.properties.sensitive.aws.kms
+
+import com.amazonaws.auth.PropertiesCredentials
+import com.amazonaws.services.kms.AWSKMSClient
+import com.amazonaws.services.kms.AWSKMSClientBuilder
+import com.amazonaws.services.kms.model.CreateAliasRequest
+import com.amazonaws.services.kms.model.CreateKeyRequest
+import com.amazonaws.services.kms.model.CreateKeyResult
+import com.amazonaws.services.kms.model.DescribeKeyRequest
+import com.amazonaws.services.kms.model.DescribeKeyResult
+import com.amazonaws.services.kms.model.GenerateDataKeyRequest
+import com.amazonaws.services.kms.model.GenerateDataKeyResult
+import com.amazonaws.services.kms.model.ScheduleKeyDeletionRequest
+import org.apache.nifi.properties.StandardNiFiProperties
+import org.apache.nifi.properties.sensitive.ProtectedNiFiProperties
+import org.apache.nifi.properties.sensitive.SensitivePropertyProtectionException
+import org.apache.nifi.properties.sensitive.SensitivePropertyProvider
+import org.apache.nifi.util.NiFiProperties
+import org.junit.After
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.security.SecureRandom
+
+
+@RunWith(JUnit4.class)
+class AWSKMSSensitivePropertyProviderIT extends GroovyTestCase {
+    private static final Logger logger = LoggerFactory.getLogger(AWSKMSSensitivePropertyProviderIT.class)
+    protected final static String CREDENTIALS_FILE = System.getProperty("user.home") + "/aws-credentials.properties";
+    private static String[] knownGoodKeys = []
+    private static AWSKMSClient client
+
+    /**
+     * This method creates a CMK, DEK, and an alias to that DEK for exercising the AWS KMS calls.
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    static void setUpOnce() throws Exception {
+        final FileInputStream fis
+        try {
+            fis = new FileInputStream(CREDENTIALS_FILE)
+        } catch (FileNotFoundException e1) {
+            fail("Could not open credentials file " + CREDENTIALS_FILE + ": " + e1.getLocalizedMessage());
+            return
+        }
+        final PropertiesCredentials credentials = new PropertiesCredentials(fis)
+
+        // We're copying the properties directly so the standard builder works.
+        System.setProperty("aws.accessKeyId", credentials.AWSAccessKeyId)
+        System.setProperty("aws.secretKey", credentials.AWSSecretKey)
+        System.setProperty("aws.region", "us-east-2")
+
+        client = AWSKMSClientBuilder.standard().build() as AWSKMSClient
+
+        // generate a cmk
+        CreateKeyRequest cmkRequest = new CreateKeyRequest().withDescription("CMK for unit tests")
+        CreateKeyResult cmkResult = client.createKey(cmkRequest)
+
```

Comment:
I would add more logging around the process occurring here to illustrate progress.

Reply:
Added additional logging with descriptive text to illustrate the AWS key construction process.